lib/index.js

```diff
@@ -104,23 +104,18 @@ internals.implementation = function(server, options) {
       scheme: scheme,
       attributes: attributes,
     };
-    let _errorType = errorType; // copies of params
-    let _message = message; // so we can over-write them below
-    let _scheme = scheme; // without a linter warning
-    let _attributes = attributes; // if you know a better way please PR!
 
-    if (options.errorFunc && internals.isFunction(options.errorFunc)) {
+    if (internals.isFunction(options.errorFunc)) {
       errorContext = options.errorFunc(errorContext);
-
-      /* istanbul ignore else */
-      if (errorContext) {
-        // if errorContext is undefined don't set attributes!
-        _errorType = errorContext.errorType;
-        _message = errorContext.message;
-        _scheme = errorContext.scheme;
-        _attributes = errorContext.attributes;
-      }
     }
+    // Since it is clearly speacified in the docs that
+    // the errorFunc must return an object with keys:
+    // errorType and message, we need not worry about
+    // errorContext being undefined
+    const _errorType = errorContext.errorType;
+    const _message = errorContext.message;
+    const _scheme = errorContext.scheme;
+    const _attributes = errorContext.attributes;
```

Comment:
Why define four `const`? if we're trying to improve/optimise this 
could we simply inline them in the `return` statement and avoid using up memory?